src/CheckoutSdk/Webhooks/RegisterWebhookRequest.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+
+namespace Checkout.Webhooks
+{
+    /// <summary>
+    /// Defines a <see cref="RegisterWebhookRequest"/>.
+    /// </summary>
+    public sealed class RegisterWebhookRequest : Webhook
+    {
+        /// <summary>
+        /// Creates a new <see cref="RegisterWebhookRequest"/> instance.
+        /// This class is suitable for registering a new webhook.
+        /// </summary>
+        /// <param name="webhook">The base webhook.</param>
+        public RegisterWebhookRequest(IWebhook webhook)
+        {
+            if (string.IsNullOrEmpty(webhook.Url)) throw new ArgumentNullException("On registration, the URL is required.", nameof(webhook.Url));
+            if (webhook.EventTypes.Count < 1) throw new ArgumentNullException("On registration, at least one event type is required.", nameof(webhook.EventTypes));
```

Comment:
```suggestion
            if (string.IsNullOrEmpty(webhook.Url)) throw new ArgumentNullException(nameof(webhook.Url), "On registration, the URL is required.");
            if (webhook.EventTypes.Count < 1) throw new ArgumentNullException(nameof(webhook.EventTypes), "On registration, at least one event type is required.");
```
Switch the order of the constructor arguments into `ArgumentNullException` so it goes `ArgumentNullException(paramName, message)`.